source/backend/routes/users.js

```diff
@@ -86,4 +88,45 @@ router.delete('/:savedRecipeId', async (req, res) => {
   }
 });
 
+/* POST /auth/signup */
+/* Creates new user in database with username, password and email -> unique user/email 
+required */
+router.post('/auth/signup', async (req, res) => {
+  try{
+    const { username } = req.username;
+    const { password } = req.password;
+    const hashedPwd = await bcrypt.hash(password, 10);
+    const { email } = req.email;
+
+    /* check username and email are unique */
+    const checkUser = await usersModel.getByUsername(username);
+    const checkEmail = await usersModel.getByEmail(email);
+
+    if (checkUser.length != 0) {
+      // not sure if status number is correct, 409 = conflict
+      return res.status(409).json({msg: "Username already exists"})
+    }
+    if ( checkEmail.length != 0) {
+      // not sure if status number is correct, 409 = conflict
+      return res.status(409).json({msg: "E-mail already in use"})
+    }
+
+    /* TODO: create new user */ 
+    const newUser = req.body.user;
```

Comment:
Line 115 should be fine, but how is our data going to be passed in the body? Will it be: 
```
{
// Option A
     "user": {
             "username": "testuser",
              "password": "badpassword",
              "email": "fake@fake.com"
      }
{
```
or will it be: 
```
{
// Option B
    "username": "testuser",
    "password": "badpassword",
    "email": "fake@fake.com"
}
```

This is a minor design detail that we should consider and set as precedence because when we implement the front end, we want to make sure to send the data properly. If we pick option A, then the way you have it is fine, if we pick option B, then we just need to make a minor change `const newUser = req.body` which is a very simple change. I'm not entirely sure which option is better for implementation, but I just want to double check it.

Reply:
+1